client/my-sites/plan-features-2023-grid/components/storage-add-on-dropdown.tsx

```diff
@@ -1,19 +1,61 @@
-import { WpcomPlansUI } from '@automattic/data-stores';
+import { AddOnMeta, WpcomPlansUI } from '@automattic/data-stores';
 import { CustomSelectControl } from '@wordpress/components';
 import { useDispatch, useSelect } from '@wordpress/data';
 import { TranslateResult, useTranslate } from 'i18n-calypso';
+import { usePlansGridContext } from '../grid-context';
 import { getStorageStringFromFeature } from '../util';
 import type { PlanSlug, StorageOption, WPComStorageAddOnSlug } from '@automattic/calypso-products';
 
 type StorageAddOnDropdownProps = {
+	label?: string;
 	planSlug: PlanSlug;
 	storageOptions: StorageOption[];
+	showCost?: boolean;
 };
 
-export const StorageAddOnDropdown = ( { planSlug, storageOptions }: StorageAddOnDropdownProps ) => {
+type StorageAddOnOptionProps = {
+	title: string;
+	cost: string | undefined;
+	showCost: boolean;
+};
+
+const getStorageOptionPrice = (
+	storageAddOnsForPlan: ( AddOnMeta | null )[] | null,
+	storageOptionSlug: string
+) => {
+	return storageAddOnsForPlan?.find( ( addOn ) => {
+		return addOn?.featureSlugs?.includes( storageOptionSlug );
+	} )?.prices?.formattedMonthlyPrice;
+};
```

Comment:
Non-blocker: I think this is another strong sign that something is off in our data representation. Generally speaking, this function accepts a list of storage add-ons, and returns the price of a given storage add-on feature by a linear search. Doing so feels quite reverse: from my intuition, the dropdown should pick an add-on product and then we can tell the price directly from the product data. 

I didn't have a chance to look into if it's possible to do it this way given the current data representation, hence noting this as a non-blocker. If it's not feasible to address it now, it should surely be addressed when we rearchitect the data.